Movie/MoviesMVP/MovieListViewController/Presenter/MovieListPresenter.swift

```diff
@@ -2,3 +2,68 @@
 // Copyright © RoadMap. All rights reserved.
 
 import Foundation
+
+/// Presenter for MovieList ViewController
+final class MovieListPresenter: MovieListPresenterProtocol {
+    // MARK: Public Properties
+
+    let networkService: NetworkServiceProtocol?
+    let router: RouterProtocol
+    var movies: [Movie]?
+    weak var view: MovieListViewProtocol?
+
+    // MARK: Initializers
+
+    init(
+        view: MovieListViewProtocol,
+        networkService: NetworkServiceProtocol,
+        router: RouterProtocol
+    ) {
+        self.view = view
+        self.networkService = networkService
+        self.router = router
+        getMovies(index: 0)
+    }
+
+    // MARK: Public Methods
+
+    func showDescription(movie: Movie) {
+        router.pushMovieDescriptionViewController(movie: movie)
+    }
+
+    func getMovies(index: Int) {
+        switch index {
+        case 0:
+            networkService?.loadPopularMovies { [weak self] moviesResults in
+                switch moviesResults {
+                case let .success(movies):
+                    self?.movies = movies
+                case let .failure(error):
+                    print(error.localizedDescription)
+                }
+                self?.view?.updateView()
```

Comment:
```suggestion
                self?.view?.updateView()
```
update надо делать в тот момент когда success прийдет а не раньше